frontend/app/src/util/AppNavigation.ts

```diff
@@ -140,6 +152,131 @@ export class V1Strategy {
       ) ?? this.appPages[0]
     )
   }
+
+  clearPageElements(
+    _elements: AppRoot,
+    mainScriptHash: string,
+    sidebarElements: BlockNode | undefined
+  ): AppRoot {
+    return AppRoot.empty(mainScriptHash, false, sidebarElements)
+  }
+}
+
+export class StrategyV2 {
```

Comment:
Should a common interface be defined here so that we can ensure `StrategyV1` and `StrategyV2` both comply to it? It seems like it'd be easy in the future to have them desync in ways that slip by code review + CI without one.

Reply:
I was thinking that, but the types were `StrategyV1 | StrategyV2`, so theoretically this must comply with the interface for both, but let me know if you think strongly.